doc/nrf/libraries/zigbee/lib_zigbee_app_helper_functions.rst

```diff
@@ -0,0 +1,322 @@
+For example, when the application wants to use the procedure of the initiator of finding & binding or use the production configuration feature.
+
+If the application wants to change or extend the default behavior, use a switch-case statement and then the default signal handler inside the default case:
+
+.. code-block:: c
+
+    void zboss_signal_handler(zb_bufid_t bufid)
+    {
+        zb_zdo_app_signal_hdr_t  * p_sg_p = NULL;
+        zb_zdo_app_signal_type_t   sig    = zb_get_app_signal(bufid, &p_sg_p);
+        zb_ret_t                   status = ZB_GET_APP_SIGNAL_STATUS(bufid);
+
+        switch (sig)
+        {
+            case ZB_BDB_SIGNAL_DEVICE_REBOOT:
+                /* fall-through */
+            case ZB_BDB_SIGNAL_STEERING:
+                if (status == RET_OK)
+                {
+                    NRF_LOG_INFO("Joined network successfully");
+                    // TODO: Start application-specific logic that requires the device to be connected to a Zigbee network.
+                }
+                else
+                {
+                    zb_bool_t comm_status;
+
+                    NRF_LOG_INFO("Unable to join the network. Restart network steering.");
+                    comm_status = bdb_start_top_level_commissioning(ZB_BDB_NETWORK_STEERING);
+                    ZB_COMM_STATUS_CHECK(comm_status);
+                }
+                break;
+
+            default:
+                // Call default signal handler.
+                ZB_ERROR_CHECK(zigbee_default_signal_handler(bufid));
+                break;
+        }
+
+        if (bufid)
+        {
+            zb_buf_free(bufid);
+        }
+    }
+
+
+.. _zarco_signal_handler_startup:
+
+Zigbee stack startup without autostart
+======================================
+
+Whenever :cpp:func:`zigbee_default_signal_handler()` is used, the stack should be started using a call to :cpp:func:`zboss_start_no_autostart()`:
+
+.. code-block:: c
+
+    zboss_start_no_autostart();
+
+The reason for this is that although the Zigbee stack provides some automatic initialization procedure with :cpp:func:`zboss_start()`, the default signal handler requires the application to start the stack without those mechanisms.
+
+Starting the stack with :cpp:func:`zboss_start_no_autostart()` has the following advantages:
+
+* It allows you to ensure better granularity in customizing the default procedure.
+* You gain a better understanding of procedures executed by the stack, because all procedures specified by the Zigbee Base Device Behavior specification are explicitly started inside the default signal handler and generate a separate signal once finished.
+
+.. figure:: /images/zigbee_signal_handler_00_stack_init.png
+   :alt: Zigbee stack startup without autostart
+
+   Zigbee stack startup without autostart
+
+When the stack is started through :cpp:func:`zboss_start_no_autostart()`, the stack generates the following signals:
```

Comment:
Rephrase "When the stack is started through `zigbee_enable()`, the stack generates the following signals:"